app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -968,11 +969,30 @@ class HomeFragment : Fragment() {
                                 text = resources.getString(R.string.wallpaper_select_error_snackbar_message),
                             )
                         }
+                        // If setting a wallpaper failed reset also the contrasting text color.
+                        requireContext().settings().currentWallpaperTextColor = 0L
                         lastAppliedWallpaperName = Wallpaper.defaultName
                     }
                 }
             }
         }
+        // Logo color should be updated in all cases.
+        applyWallpaperTextColor()
+    }
+
+    /**
+     * Apply a color better contrasting with the current wallpaper to the Fenix logo and private mode switcher.
+     */
+    private fun applyWallpaperTextColor() {
+        if (shouldEnableWallpaper()) {
+            val tintColor = when (val color = requireContext().settings().currentWallpaperTextColor.toInt()) {
+                0 -> null // a null ColorStateList will clear the current tint
+                else -> ColorStateList.valueOf(color)
+            }
```

Comment:
Interesting that `ColoStateList` requires an `Int` but other hex values expect `Long`s. I'm not sure there's a good way for us to avoid any errors in the conversion outside of relying on QA